app/angular/src/server/config.js

```diff
@@ -1,97 +1,37 @@
-  const storybookCustomAddonsPath = path.resolve(configDir, 'addons.js');
-  if (fs.existsSync(storybookCustomAddonsPath)) {
-    logger.info('=> Loading custom addons config.');
-    config.entry.manager.unshift(storybookCustomAddonsPath);
-  }
+let cliWebpackConfigOptions;
 
-  // Check whether project has Angular CLI configuration file
-  const cliWebpackConfigOptions = getAngularCliWebpackConfigOptions(process.cwd());
-  if (cliWebpackConfigOptions) {
-    logger.info('=> Loading angular-cli config.');
-  }
+const configLoader = configLoaderCreator({
+  defaultConfigName: 'angular-cli',
+  defaultBabelConfig: defaultConfig,
+  wrapInitialConfig: (config, configDir) => {
+    const tsRule = config.module.rules[1];
 
-  const defaultConfig = applyAngularCliWebpackConfig(
-    createDefaultWebpackConfig(config),
-    cliWebpackConfigOptions
-  );
+    tsRule.loaders[0].options = loadTsConfig(configDir);
 
-  // Check whether user has a custom webpack config file and
-  // return the (extended) base configuration if it's not available.
-  const customConfigPath = path.resolve(configDir, 'webpack.config.js');
+    return config;
+  },
+  wrapDefaultConfig: config => {
+    // Check whether project has Angular CLI configuration file
+    cliWebpackConfigOptions = getAngularCliWebpackConfigOptions(process.cwd());
 
-  if (!fs.existsSync(customConfigPath)) {
-    logger.info('=> Using default webpack setup based on "angular-cli".');
-    return defaultConfig;
-  }
-  const customConfig = require(customConfigPath);
+    if (cliWebpackConfigOptions) {
+      logger.info('=> Loading angular-cli config.');
+    }
 
-  if (typeof customConfig === 'function') {
-    logger.info('=> Loading custom webpack config (full-control mode).');
-    return customConfig(
-      applyAngularCliWebpackConfig(config, cliWebpackConfigOptions),
-      configType,
-      defaultConfig
-    );
-  }
-  logger.info('=> Loading custom webpack config (extending mode).');
+    return applyAngularCliWebpackConfig(config, cliWebpackConfigOptions);
+  },
+  wrapBasicConfig: config => applyAngularCliWebpackConfig(config, cliWebpackConfigOptions),
+});
 
-  return {
-    ...customConfig,
-    // We'll always load our configurations after the custom config.
-    // So, we'll always load the stuff we need.
-    ...config,
-    // Override with custom devtool if provided
-    devtool: customConfig.devtool || config.devtool,
-    // We need to use our and custom plugins.
-    plugins: [...config.plugins, ...(customConfig.plugins || [])],
-    module: {
-      ...config.module,
-      // We need to use our and custom rules.
-      ...customConfig.module,
-      rules: [...config.module.rules, ...(customConfig.module.rules || [])],
-    },
-    resolve: {
-      ...config.resolve,
-      ...customConfig.resolve,
-      alias: {
-        ...config.alias,
-        ...(customConfig.resolve && customConfig.resolve.alias),
-      },
-    },
-  };
-}
```

Comment:
I just realized that we don't apply angular cli config to extend mode. Are there any good reasons for that?

Reply:
I don't remember already 😓

Reply:
@danielduan , what do you mean?